src/ur/ur_driver.cpp

```diff
@@ -223,6 +232,78 @@ bool UrDriver::writeTrajectoryControlMessage(const control::TrajectoryControlMes
   return reverse_interface_->writeTrajectoryControlMessage(trajectory_action, point_number);
 }
 
+bool UrDriver::zeroFTSensor()
+{
+  if (getVersion().major < 5)
+  {
+    std::stringstream ss;
+    ss << "Zeroing the Force-Torque sensor is only available for e-Series robots (Major version >= 5). This robot's "
+          "version is "
+       << getVersion();
+    URCL_LOG_ERROR(ss.str().c_str());
+    return false;
+  }
+  else
+  {
+    if (script_command_interface_->clientConnected())
+    {
+      return script_command_interface_->zeroFTSensor();
+    }
+    else
+    {
+      std::stringstream cmd;
+      cmd << "sec tareSetup():" << std::endl << " zero_ftsensor()" << std::endl << "end";
+      return sendScript(cmd.str());
+    }
+  }
+}
+
+bool UrDriver::setPayload(const float mass, const vector3d_t& cog)
+{
+  if (script_command_interface_->clientConnected())
+  {
+    return script_command_interface_->setPayload(mass, &cog);
+  }
+  else
+  {
+    std::stringstream cmd;
+    cmd.imbue(std::locale::classic());  // Make sure, decimal divider is actually '.'
+    cmd << "sec setup():" << std::endl
+        << " set_payload(" << mass << ", [" << cog[0] << ", " << cog[1] << ", " << cog[2] << "])" << std::endl
+        << "end";
+    return sendScript(cmd.str());
+  }
+}
+
+bool UrDriver::setToolVoltage(const ToolVoltage voltage)
+{
+  // Test that the tool voltage is either 0, 12 or 24.
+  switch (voltage)
+  {
+    case ToolVoltage::OFF:
+      break;
+    case ToolVoltage::_12V:
+      break;
+    case ToolVoltage::_24V:
+      break;
+    default:
+      std::stringstream ss;
+      ss << "The tool voltage should be 0, 12 or 24. The tool voltage is " << toUnderlying(voltage);
+      throw UrException(ss.str().c_str());
```

Comment:
I'm not remembering completely, whether we catch all exception types inside the driver. Having an `invalid_argument`  Exception seems more appropriate to me, but in general: It might not necessarily be the best choice to throw an exception here? We are inside a boolean function, so simply printing an error and returning `false` might also be an appropriate way of handling this, don't you think?

Reply:
I agree that it would be better to print an error message and return false.